test/integration/ctl/ctl_create_cr_test.go

```diff
@@ -293,22 +297,29 @@ func TestCtlCreateCRSuccessful(t *testing.T) {
 				// by the CLI command yet.
 				// Once it has been created, set the `status.certificate` and `Ready` condition so that the `--fetch-certificate`
 				// part of the command is able to proceed.
+				doneCh := make(chan struct{})
+				pollCtx, cancelPoll := context.WithCancel(ctx)
+				defer func() {
+					cancelPoll()
+					<-doneCh
```

Comment:
cancel the `pollCtx`, so that the goroutine ends and wait for the routine to end using doneCh.